### PR TITLE
Misc fixes

### LINKED ISSUE #1418: reconciler: incomplete log output
this line 
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/d60419e4f0063023ea3953c25336c0b2ee9eb9f8/pkg/controllers/reconciler.go#L233

is causing incomplete log lines that look suspicious especially when the diff is empty:
```
I0515 14:13:19.142299 1922029 reconciler.go:233] "" controller="dsadeviceplugin" controllerGroup="deviceplugin.intel.com" controllerKind="DsaDevicePlugin" DsaDevicePlugin="dsadeviceplugin-test" namespace="" name=
"dsadeviceplugin-test" reconcileID=d9fa9595-5651-4796-9c6d-75073106860e ="(MISSING)"
```

### LINKED ISSUE #1483: fix IAA README for demos
There is no `iaa-qpl-demo` image to build anymore.

### LINKED ISSUE #1556: gpu: nfd platform rules have empty keys
Initially found in the helm charts: https://github.com/intel/helm-charts/issues/41

https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/deployments/nfd/overlays/node-feature-rules/platform-labeling-rules.yaml#L19

### PR DESCRIPTION
**Pull request by tkatila**
Fixes #1418 
Fixes #1556 
Fixes #1483 
And bumps the operator min version.

### CODE REVIEW

**Comment by tkatila** [comment]
Tested dsa/iaa initcontainer changes some more, worked as expected. Ready for review.

File: pkg/controllers/iaa/controller.go [hunk 1]
```diff
@@ -219,9 +219,12 @@ func provisioningUpdate(ds *apps.DaemonSet, dp *devicepluginv1.IaaDevicePlugin)
 	found := false
 
 	for _, container := range ds.Spec.Template.Spec.InitContainers {
-		if container.Name == "intel-iaa-initcontainer" && container.Image != dp.Spec.InitImage {
+		if container.Name == "intel-iaa-initcontainer" {
```
Comment (mythi) [inline pkg/controllers/iaa/controller.go:222]: our code has this minor inconsistency that the `const` is not used all over the place (also, it has a typo?) can you check other places too:
```suggestion
		if container.Name == initcontainerName {
```

File: pkg/controllers/gpu/controller.go [hunk 2]
```diff
@@ -242,32 +242,54 @@ func removeVolumeMount(volumeMounts []v1.VolumeMount, name string) []v1.VolumeMo
...
 }
 
-func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (updated bool) {
-	dp := rawObj.(*devicepluginv1.GpuDevicePlugin)
-
-	if ds.Spec.Template.Spec.Containers[0].Image != dp.Spec.Image {
-		ds.Spec.Template.Spec.Containers[0].Image = dp.Spec.Image
-		updated = true
-	}
+func processInitcontainer(ds *apps.DaemonSet, dp *devicepluginv1.GpuDevicePlugin) bool {
```
Comment (hj-johannes-lee) [inline pkg/controllers/gpu/controller.go:245]: nitpick: Other parts use 'initContainer'. Do we need to unify them?

Comment (tkatila) [reply pkg/controllers/gpu/controller.go]: Unified.

File: pkg/controllers/iaa/controller.go (hunk 1 above)
Comment (tkatila) [reply pkg/controllers/iaa/controller.go]: Added more `const`s and fixed the typo.

**Review by mythi — APPROVED** [review]

**Pull request merged**